src/plat/pc99/machine/hardware.c

```diff
@@ -17,16 +17,47 @@
 
 BOOT_CODE bool_t platAddDevices(void)
 {
-    /* remove the MSI region as poking at this is undefined and may allow for
-     * the user to generate arbitrary MSI interrupts. Only need to consider
-     * this if it would actually be in the user device region */
-    if (CONFIG_PADDR_USER_DEVICE_TOP > 0xFFFFFFF8) {
-        if (!reserve_region((p_region_t) {
-        (word_t)0xFFFFFFF8, (word_t)0xFFFFFFF8 + 8
-        })) {
+
+#ifdef CONFIG_ARCH_IA32
+
+    /* The last 64 KiB [0xffff0000 - ffffffff] are reserved and can't be used.
+     * ToDo: add some details, why actually?
+     */
+    p_region_t p_reg_reserved = {
```

Comment:
I wonder, can't reserve this memory in the Python scrips already via a device tree overlay. Or maybe add this in the config and then have the script pick up some reserved regions from there.